packages/webapp/src/components/Navigation/index.jsx

```diff
@@ -19,53 +21,85 @@ export default function PureNavigation({
   history,
   isFarmSelected,
   hidden,
+  children,
 }) {
   // Side Drawer
-  const [isSidebarOpen, setIsSidebarOpen] = useState(false);
-  const openSidebar = () => setIsSidebarOpen(true);
-  const closeSidebar = () => setIsSidebarOpen(false);
+  const [isSideMenuOpen, setIsSideMenuOpen] = useState(false);
+  const [isCompactSideMenu, setIsCompactSideMenu] = useState(false);
+  const [hasSideMenuBeenExpanded, setHasSideMenuBeenExpanded] = useState(false);
+
+  const toggleSideMenu = () => {
+    setHasSideMenuBeenExpanded(isCompactSideMenu);
+    setIsCompactSideMenu(!isCompactSideMenu);
+  };
+
+  const openSideMenu = () => setIsSideMenuOpen(true);
+  const closeSideMenu = () => setIsSideMenuOpen(false);
   const theme = useTheme();
   const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
 
-  return (
-    !hidden && (
-      <>
+  return hidden ? (
+    children
+  ) : (
+    <>
+      {isFarmSelected && (
+        <>
+          {/* Sidebar Menu */}
+          {isMobile ? (
```

Comment:
I know I am not assigned to this PR but wanted to take a peek! The animation is really nice looking. 

I just saw you move this in (nice!) and thought the conditional was getting big here. The decision to use a drawer or not might be neatly handled in the SideMenu similarly to Top.

Reply:
Done! Can you re-test and see if everything still looks good? Worried I might have messed something up in the refactor 😅

Reply:
Love it! Thats some nice stuff right there!